library/Garden/Web/Data.php

```diff
@@ -0,0 +1,166 @@
+        }
+        return isset($this->data[$name]) ? $this->data[$name] : $default;
+    }
+
+    /**
+     * Set a single item to the data array.
+     *
+     * @param string $name The key to set.
+     * @param mixed $value The new value.
+     * @return $this
+     */
+    public function setDataItem($name, $value) {
+        if (!is_array($this->data) && !($this->data instanceof \ArrayAccess)) {
+            throw new \Exception("Data is not an array.", 500);
+        }
+        $this->data[$name] = $value;
+        return $this;
+    }
+
+    /**
+     * Get the entire data payload.
+     *
+     * @return mixed Returns the data.
+     */
+    public function getData() {
+        return $this->data;
+    }
+
+    /**
+     * Set the entire data payload.
+     *
+     * @param mixed $data The new data array.
+     * @return $this
+     */
+    public function setData($data) {
+        $this->data = $data;
+        return $this;
+    }
+
+    /**
+     * Get the HTTP status.
+     *
+     * @return int Returns the status.
+     */
+    public function getStatus() {
+        return $this->getMetaItem('status', 200);
+    }
+
+    /**
+     * Set the HTTP status.
+     *
+     * @param int $status The new status.
+     * @return $this
+     */
+    public function setStatus($status) {
+        return $this->setMetaItem('status', $status);
+    }
+
+    /**
+     * Get a single item from the meta array.
+     *
+     * @param string $name The key to get from.
+     * @param mixed $default The default value if no item at the key exists.
+     * @return mixed Returns the meta value.
+     */
+    public function getMetaItem($name, $default = null) {
+        return isset($this->meta[$name]) ? $this->meta[$name] : $default;
+    }
+
+    /**
+     * Set a single item to the meta array.
+     *
+     * @param string $name The key to set.
+     * @param mixed $value The new value.
+     * @return $this
+     */
+    public function setMetaItem($name, $value) {
+        $this->meta[$name] = $value;
+        return $this;
+    }
+
+    /**
+     * Get the entire meta array.
+     *
+     * @return array Returns the meta.
+     */
+    public function getMeta() {
+        return $this->meta;
+    }
+
+    /**
+     * Set the entire meta array.
+     *
+     * @param array $meta The new meta array.
+     * @return $this
+     */
+    public function setMeta(array $meta) {
+        $this->meta = $meta;
+        return $this;
+    }
+
+    /**
+     * Specify data which should be serialized to JSON.
+     *
+     * @return mixed data which can be serialized by <b>json_encode</b>,
+     * which is a value of any type other than a resource.
+     * @link http://php.net/manual/en/jsonserializable.jsonserialize.php
+     */
+    public function jsonSerialize() {
+        return $this->getData();
+    }
+
+    /**
+     * Render the response to the output.
+     */
+    public function render() {
+        header('', true, $this->getStatus());
```

Comment:
Would [`http_response_code`](http://php.net/manual/en/function.http-response-code.php) be more appropriate here? The [docs for `header`](http://php.net/manual/en/function.header.php#refsect1-function.header-parameters) state the response code parameter "only has an effect if the string is *not empty*".